### DISCUSSION TITLE
Mathematical Inconsistency: Running Statistics notation and variance calculation

### DISCUSSION THREAD

**Discussion by debemdeboas** [online-ml/river#1626] [General]
In the Running Statistics subsection of the documentation ([From batch to online/stream](https://riverml.xyz/latest/examples/batch-to-online/#from-batch-to-onlinestream)), there appears to be a mathematical inconsistency in how variance is presented.

Current notation:

$$
\begin{cases}
n_{t+1} = n_t + 1 \\
\mu_{t+1} = \mu_t + \frac{x - \mu_t}{n_{t+1}} \\
s_{t+1} = s_t + (x - \mu_t) \times (x - \mu_{t+1}) \\
\sigma_{t+1} = \frac{s_{t+1}}{n_{t+1}}
\end{cases}
$$

The text states that " $\sigma_t$ is the running variance at time $t$", but this is incorrect.
The quantity $\frac{s_{t+1}}{n_{t+1}}$ where s is the running sum of squares represents neither the variance nor the standard deviation - it's simply the sum of squares divided by n.

Proposed correction:
1. Either rename $\sigma_t$ to something like ss_t to indicate it's a scaled sum of squares
2. Or modify the formula to properly show variance calculation by adding appropriate scaling

**Comment by debemdeboas** [comment]
I may be wrong about this too. Statistics is not my strong suit.

**Reply by smastelini** [reply]
Hi @debemdeboas. The formulae presented above refers to the Welford's algorithm (given the denominator, the degree of freedom is zero) . If you want to check a bit more about why this strategy is used rather than the the sum of squares one, you can check this [presentation](https://smastelini.github.io/pdf/qo-waikato.pdf) I made quite some time ago :D
